conformance/lang/expressions/logical-expressions/logical_and.balt

```diff
@@ -0,0 +1,732 @@
+    boolean _ =  true && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  false && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (a == "text"? 2 : 1) && false; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (a == "text"? 2 : 1) && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using checking expression.
+Labels: logical-expr, checking-expr
+
+function init() {
+    boolean _ =  true && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  check int:fromString("123") && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  check int:fromString("123") && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using trap expression.
+Labels: logical-expr, trap-expr
+
+function init() {
+    boolean _ =  true && trap divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+}
+
+function divideNum(int a, int b) returns int {
+    return a / b;
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml navigate expression.
+Labels: logical-expr, xml-navigate-expr
+
+function init() {
+    xml a = xml `<root><ele>true</ele><ele>false</ele></root>`;
+    boolean _ =  true && a/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/* && a/*; // @error static type of each expression must be a subtype of boolean
+    
+    boolean _ =  true && a/<ele>; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele> && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele> && a/<ele>; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ =  true && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele>/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele>/* && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+}
```

Comment:
For future tests, we don't need this many error tests with different kinds of expressions. It is enough to have several error tests with different expression kinds.